dcmdump/src/main.rs

```diff
@@ -27,6 +29,42 @@ const ERROR_READ: i32 = -2;
 /// Exit code for when an error emerged while dumping the file.
 const ERROR_PRINT: i32 = -3;
 
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+enum DumpValue<T>
+where
+    T: ToString,
+{
+    TagNum(T),
+    Alias(T),
+    Num(T),
+    Str(T),
+    DateTime(T),
+    Invalid(T),
+    Nothing,
+}
+
+impl<T> fmt::Display for DumpValue<T>
+where
+    T: ToString,
```

Comment:
I feel that one could avoid constructing heap allocated strings here, but we can leave that for another time if it requires too many changes.

Reply:
I'm interested in doing this, but I'm not sure how to go about it so it may make sense for a different issue? Not entirely sure how to convert some of the non-string types to `&str` or `String` without allocating in this context, but I'm likely missing something

Reply:
This could well be a limitation of `colored`. The struct `ColoredString` works with strings internally, whereas a possible implementation of a hypothetical _`ColorisedSomething`_ could work for any type implementing `Display`, without allocating a string. Sounds like an opportunity to take this concern to the respective [repository](https://github.com/mackwic/colored).